systemtests/src/test/java/io/enmasse/systemtest/isolated/CommonTest.java

```diff
@@ -425,6 +452,258 @@ void testMessagingDuringRestartComponents() throws Exception {
+                .withInfraConfigRef(testInfra.getMetadata().getName())
+                .withResourceLimits(Arrays.asList(
+                        new ResourceAllowance("broker", 1.0))
+                        .stream().collect(Collectors.toMap(ResourceAllowance::getName, ResourceAllowance::getMax)))
+                .withAddressPlans(Arrays.asList(exampleAddressPlan.getMetadata().getName()))
+                .endSpec()
+                .build();
+        resourcesManager.createAddressSpacePlan(exampleSpacePlan);
+
+        AddressSpace exampleAddressSpace = new AddressSpaceBuilder()
+                .withNewMetadata()
+                .withName("test-address-space-keeps-operable")
+                .withNamespace(kubernetes.getInfraNamespace())
+                .endMetadata()
+                .withNewSpec()
+                .withType(AddressSpaceType.BROKERED.toString())
+                .withPlan(exampleSpacePlan.getMetadata().getName())
+                .withNewAuthenticationService()
+                .withName("standard-authservice")
+                .endAuthenticationService()
+                .endSpec()
+                .build();
+
+        resourcesManager.createAddressSpace(exampleAddressSpace);
+
+        doTestBrokerRemainsOperableOnFullAddress(exampleAddressPlan, exampleAddressSpace, false, false);
+
+    }
+
+    private void doTestBrokerRemainsOperableOnFullAddress(AddressPlan addressPlan, AddressSpace addressSpace, boolean retry, boolean createSecondAddressAfter) throws Exception {
+        UserCredentials user = new UserCredentials("dummy", "supersecure");
+        resourcesManager.createOrUpdateUser(addressSpace, user);
+
+        var address1 = new AddressBuilder()
+                .withNewMetadata()
+                .withNamespace(kubernetes.getInfraNamespace())
+                .withName(AddressUtils.generateAddressMetadataName(addressSpace, "example-queue-1"))
+                .endMetadata()
+                .withNewSpec()
+                .withType(AddressType.QUEUE.toString())
+                .withAddress("example-queue-1")
+                .withPlan(addressPlan.getMetadata().getName())
+                .endSpec()
+                .build();
+
+        var address2 = new AddressBuilder()
+                .withNewMetadata()
+                .withNamespace(kubernetes.getInfraNamespace())
+                .withName(AddressUtils.generateAddressMetadataName(addressSpace, "example-queue-2"))
+                .endMetadata()
+                .withNewSpec()
+                .withType(AddressType.QUEUE.toString())
+                .withAddress("example-queue-2")
+                .withPlan(addressPlan.getMetadata().getName())
+                .endSpec()
+                .build();
+
+        if (createSecondAddressAfter) {
+            resourcesManager.setAddresses(address1);
+        } else {
+            resourcesManager.setAddresses(address1, address2);
+        }
+
```

Comment:
You've got an implicit assumption that the two queues are on the same broker (single). Given the state of the plans, this assumption should be safe, but I think the test should verify that starting condition is true.   That way, if we ever change the way the queue to broker allocation works, this will fail automatically.    In short, I am asking that you verify that the two queues have the same status.brokerStatuses.

Reply:
great suggestion, I'll add it, thanks